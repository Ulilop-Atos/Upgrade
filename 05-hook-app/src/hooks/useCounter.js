import { useState } from 'react';

export const useCounter = ( initialState = 0 ) => {

  const [state, setState] = useState(initialState);
  
  const increment = ( n = 1 ) => {
    setState(state + n );
  };
  
  const decrement = ( n = 1 ) => {
    setState(state - n );
  };
 
  const reset = ( n = initialState ) => {
    setState( initialState );
  };

  return ({
    state,
    decrement,
    increment,
    reset,
  });
}
