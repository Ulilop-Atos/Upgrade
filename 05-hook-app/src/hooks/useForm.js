import PropTypes from 'prop-types';
import { useState } from 'react';

export const useForm = ( initialState = {} ) => {

  const [values, setValues] = useState(initialState);

  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [ target.name ]: target.value, 
    });
  }

  return [values, handleInputChange];

}

useForm.propTypes = {
  initialState: PropTypes.object,
}
