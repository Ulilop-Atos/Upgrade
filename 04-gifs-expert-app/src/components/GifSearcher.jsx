import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifItem from './GifItem';

const GifSearcher = ({ category }) => {

  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{ category }</h3>

      { loading && <p className='animate__animated animate__flash'>Loading</p> }

      <div className="card-container animate__animated animate__fadeIn">
      {
        images.map( image => <GifItem key={image.id} {...image} /> )
      }
      </div>
    </>
  );
};

GifSearcher.propTypes = {
    category: PropTypes.string.isRequired,
};

export default GifSearcher;
