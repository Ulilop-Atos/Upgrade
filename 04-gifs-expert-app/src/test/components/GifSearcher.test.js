import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifSearcher from '../../components/GifSearcher';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Tests for <GifSearcher />', () => {
  const category = 'Test';

  test('should display properly', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifSearcher category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should display images after loading hook', () => {

    const imgs =  [{
      id: 'ABC',
      url: 'https://localhost.test.url/abc.gif',
      title: 'Test gif ABC' 
    }, 
    {
      id: 'DEF',
      url: 'https://localhost.test.url/def.gif',
      title: 'Test gif DEF' 
    }];

    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false,
    });

    const wrapper = shallow(<GifSearcher category={category} />);

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifItem').length ).toBe( imgs.length );
  });


});