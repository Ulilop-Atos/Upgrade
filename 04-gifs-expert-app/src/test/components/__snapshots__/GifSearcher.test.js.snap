// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifSearcher /> should display images after loading hook 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <div
    className="card-container animate__animated animate__fadeIn"
  >
    <GifItem
      id="ABC"
      key="ABC"
      title="Test gif ABC"
      url="https://localhost.test.url/abc.gif"
    />
    <GifItem
      id="DEF"
      key="DEF"
      title="Test gif DEF"
      url="https://localhost.test.url/def.gif"
    />
  </div>
</Fragment>
`;

exports[`Tests for <GifSearcher /> should display properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-container animate__animated animate__fadeIn"
  />
</Fragment>
`;
